### ISSUE TITLE
Focus ring opacity on text input not working with arbitrary value

### ISSUE THREAD

**Issue by Noitidart** [tailwindlabs/tailwindcss#7521]
**What version of Tailwind CSS are you using?**

"tailwindcss": "^3.0.18",

**What build tool (or framework if it abstracts the build tool) are you using?**

"next": "12.0.9",

**What version of Node.js are you using?**

v12.19.0

**What browser are you using?**

All

**What operating system are you using?**

macOS

**Reproduction URL**

https://play.tailwindcss.com/Tv8lkLkkLn

**Describe your issue**

A custom ring to text input with arbritrary value works fine like this:

```
<input type="text" className="focus:ring focus:ring-[rgba(0, 244, 0)]" />
```

However it's opacity cannot be reduced with `focus:ring-opacity-30`.

Trying it, causes the arbitrary value to be ignored, and the default ring from `focus:ring` shows.

```
<input type="text" className="focus:ring focus:ring-[rgba(0, 244, 0)] focus:ring-opacity-30" />
```

Here is a great explantion by JHeth from Stackoverflow:
Tailwind arbitrary values don't work with rgb or rgba values, [this](https://github.com/tailwindlabs/tailwindcss/issues/5006#issuecomment-882229133) issue on Github exposed it and this [pull](https://github.com/tailwindlabs/tailwindcss/pull/5239) was supposed to fix it but you can still see the behavior on [Tailwind Play](https://play.tailwindcss.com/Tv8lkLkkLn) quite clearly, there is no generated class when using arbitrary rgb values (if there were, a tiny color swatch would appear next to the class).

Major thanks to JHeth from Stackoverflow for helping identify this bug and creating the Tailplay - https://stackoverflow.com/a/71168939/1828637

**Comment by adamwathan** [comment]
Yeah not possible to use the opacity utilities alongside CSS variables like that, there's just literally no way to do it with CSS because setting the opacity means changing the actual color value.

Some docs related to that:

https://tailwindcss.com/docs/customizing-colors#using-css-variables

Here's a demo where it does work, by only setting the color _channels_ in the variable instead of the full color so that the final color can be composed alongside the opacity variable:

https://play.tailwindcss.com/DZbDhtoj3f

**Comment by Noitidart** [comment]
Oh thank you very much that is very interesting!
